docs/use-cases/resources.md

```diff
@@ -0,0 +1,261 @@
+
+### Webhook
+
+```
+$ convox rack resources options webhook
+NAME  DEFAULT  DESCRIPTION
+Url            Webhook URL
+```
+
+Internally, Convox uses the `webhook` resource to enable Slack integration.  You can do similar to receive [notifications](/console/notifications) about important events happening within your apps and Rack.  
+
+# External Resources
+
+For any resource types that Convox does not natively support, it is still very easy to integrate them into your infrastructure and environments.  
+Best practice is to inject an environment variable for each external resource that you wish to integrate.  
+
+For example if you needed a MariaDB instance, you would go create that in RDS, ensure that the VPC and Security Group allows access to it from your Rack, get the appropriate endpoint, then in the application that needs access, set the env var for your connection string:
+
+```
+$ convox env set MARIADB_URL=jdbc:mariadb://mariadb-instance1.123456789012.us-east-1.rds.amazonaws.com:3306/DB?user=myUsername&password=myPassword -a myapplication -r acme/production
+```
+
+You will also need to add the `MARIADB_URL` env var to the `environment` configuration of your service in your `convox.yml`:
+
+```
+services:
+  app:
+    ...
+    environment:
+      - MARIADB_URL
+      - ...
+    ...  
+```
+
+<div class="block-callout block-show-callout type-info" markdown="1">
+As environment variables are encrypted and stored securely in KMS, your credentials are secure within your environment.  You may also route using the internal IP address if you wish, for locked down environments.
+</div>
+
+Through this, you can securely control and configure access to different resources based on app and environment.  It is even possible to integrate resources and infrastructure elements that are external to your cloud environment as long as they are accessible.
+
+- For External Resources that are still within your AWS infrastructure, you should create them within the same VPC as your Rack, within a VPC that your Rack VPC [peers with](https://docs.aws.amazon.com/vpc/latest/userguide/vpc-peering.html), or in a way that is publically accessible (with a public endpoint), depending on your security needs, and the type of Resource you are creating.  Any Security Group [configuration](https://docs.aws.amazon.com/vpc/latest/userguide/VPC_SecurityGroups.html#SecurityGroupRules) for your Resource should also allow access to your Resource's port from the Rack Security Group.
+
+- For External Resources that are outside of your AWS infrastructure, they should be publically accessible, if your Rack is public, as AWS cannot guarantee the IP address that a request will come from.  If your Rack is [private](/management/private-networking), then all outbound traffic from it will pass through a NAT Gateway with a static IP address so you may if you wish limit access to the Resource based on those IP addresses (up to 3 Gateways are created, one for each Availability Zone).
```

Comment:
Publically accessible is a bit loaded here.. my first thought was about authentication. Maybe try to rephrase this section purely in terms of IPs/networking? Something like this (but not this exactly):

> External Resource that are outside of your AWS infrastructure will need to be network-accessible to your Rack instances. For private Racks, traffic will come from the IPs of your NAT gateways (how to find them?). If your Rack is not set to private the IPs will not be easily predictable and you'll likely need to make the resource available to the public internet.